### DISCUSSION TITLE
Documentation search with AI assistant

### DISCUSSION THREAD

**Discussion by skovy** [facebook/docusaurus#9198] [Show and tell]
Hey all, having an AI assistant with access to specific data has become very popular. The primary benefit being you don't have to copy/paste content like documentation into an LLM, it can help find an answer faster and can fill in missing pieces between disparate pieces of documentation. It's not necessarily complex but does require a few pieces of infrastructure (server, database, vector store, LLM API). That's where Asktro (https://asktro.com) comes in, it's a hosted solution to unlock natural language search and an AI assistant that is specific to your docs.

You can think of it as Supabase-style docs search as a service.

It's leveraging Docusaurus's (and Nextra's) opinionated approach to ingest the documentation at build time to chunk, embed, and persist the documentation in a vector store. This allows both natural language search and an AI assistant. 

You can see a demo of it on the Docusaurus starter project here: https://docusaurus.asktro.com

Docs on how it works to integrate: https://asktro.com/docs/docusaurus

Would love to hear your thoughts and feedback, thanks!

![image](https://github.com/facebook/docusaurus/assets/5247455/925d6f8a-dc2d-4a07-96cf-64cc22420662)

**Comment by slorber** [comment]
Looks nice 👍 

We already have a other solutions proposed in this discussion: https://github.com/facebook/docusaurus/discussions/8835

**Comment by ntucker** [comment]
Is this project dead?

**Comment by jovezhong** [comment]
It'll be great to have open-source/self-hosting ask-ai plugin for docusaurus

**Comment by Ekene1998** [comment]
Awesome

**Comment by jovezhong** [comment]
Search box with AI assistant is also coming to Docsearch v4. I am waiting for the PR merge of https://github.com/facebook/docusaurus/pull/11327

**Comment by ivictbor** [comment]
At Tluma we provide free & unlimited Ask AI widget for [Docusaurus docs](https://tluma.ai/add-to-docusarus), for open-source public projects backed by GitHub. 
Installation is very simple all you need to do is add the minimal config:

```
window.tlumaConfig = {
  source: "devforth/adminforth", // public GitHub project which has your docusaurus docs , use GitHub "org/repo" or "owner/repo"
};
</script>
<script src="https://tluma.ai/widget.js" async></script>
```

This is it. Docs are always auto-indexed from default branch with a daily autoupdate.

**Comment by crawldesk** [comment]
Hey, this is really cool to see — AI-powered docs search is definitely becoming a must-have.

We’re building something similar with Crawldesk, but with a slightly different approach. Instead of relying only on build-time indexing (like Docusaurus/Nextra pipelines), Crawldesk focuses on dynamic crawling + real-time syncing of documentation.

That means:

Works with any docs site (not just Docusaurus)
No strict build-time integration needed
Can handle frequently updated or external docs
Provides an Ask AI widget trained on your content
Supports continuous re-crawling to keep answers fresh

We’ve seen a lot of teams struggle when their docs are:

spread across multiple sources
behind dynamic rendering
or updated outside the main repo

That’s where Crawldesk tries to help.

https://crawldesk.com/

https://www.youtube.com/watch?v=P85IDzEv4eo
